### ISSUE TITLE
LAST UPDATED FEATURE: Refactor Code for Various Jobs That Update Course Info

### ISSUE THREAD

**Issue by b4k3s** [ucsb-cs156-s23/proj-courses-s23-7pm-1#23]
Refactor the code for the various jobs that update course info to make it more DRY (perhaps adding either extra methods to the `ConvertedSectionCollection` class, or adding a new service class.) **I strongly encourage doing this as a separate issue* before adding in the code that updates the `UpdateCollection` for at least two reasons: (1) It can be done in parallel with the issues above, by another member of the team (2) It is much easier to code review and test a "pure refactor", since it is supposed to keep the functionality the same while restructuring the code. Often, no new tests need to be added since existing tests already cover the code and mutations to it. If this refactor is successful, afterwards, there will be only one place in the code that all jobs that update course info share, and that will make it easier to add the code to update the `updates` collection.
